snuba/utils/kafka/consumers/confluent.py

```diff
@@ -0,0 +1,66 @@
+from typing import Any, Callable, Mapping, Optional, Sequence, Tuple
+
+from confluent_kafka import Consumer as ConfluentConsumer
+from confluent_kafka import Message, TopicPartition
+
+from snuba.utils.kafka.consumers.abstract import (
+    Consumer as AbstractConsumer,
+    Topic,
+    Partition,
+    Offset,
+)
+
+
+class Consumer(AbstractConsumer[Message]):
+    def __init__(self, configuration: Mapping[str, Any]) -> None:
+        self.__consumer = ConfluentConsumer(configuration)
+
+    def __wrap_assignment_callback(
+        self, callback: Callable[[Sequence[Tuple[Topic, Partition]]], None]
+    ) -> Callable[[ConfluentConsumer, Sequence[TopicPartition]], None]:
+        """
+        Wraps a partition assignment callback with a wrapper function that
+        translates the Confluent Kafka consumer callback signature into the
+        arguments expected by the callback.
+        """
+
+        def wrapper(consumer: Consumer, assignment: Sequence[TopicPartition]) -> None:
+            callback([(Topic(i.topic), Partition(i.partition)) for i in assignment])
```

Comment:
What was the reasoning for not passing the consumer to the callback? Is it because the consumer is wrapped by this class?
Asking because it may be a bit for the someone reading this code that the wrapper around the confluent consumer has some subtile differences in the interface. 
Maybe adding a comment explaining why we drop the consumer parameter here ?

Reply:
Oops — missed this comment earlier, but the answer was here, already: https://github.com/getsentry/snuba/pull/485#discussion_r328807511

This also prevents the Confluent consumer instance itself from leaking if we were to not wrap the callback.

> Asking because it may be a bit for the someone reading this code that the wrapper around the confluent consumer has some subtile differences in the interface.

I agree this is going to be a challenge, I think the abstract class should be clearly documented, and we should point at that documentation as the canonical source rather than exposing the Confluent internals whenever possible, because I think as we've discussed in the past, the wording of their docs can be a challenge to read at times.